### DISCUSSION TITLE
Laravel Lang: Common

### DISCUSSION THREAD

**Discussion by andrey-helldar** [Laravel-Lang/common#5] [Q&A]
@Laravel-Lang/laravel-lang, hi guys!

Today I started a new project - [Laravel Lang: Common](https://github.com/Laravel-Lang/common)

It is needed in order to easily connect the necessary packages of the required versions to the application, because we actually have no relationship between them.

For example, here is one of the problems: https://github.com/Laravel-Lang/lang/discussions/2368

Actually, the problem itself lies in the fact that if we have several packaged solutions, there is no general description of how to use them, and beginners often get confused about this.

I thought to solve this problem in two steps:

1. Create a common project, by installing which it will pull the necessary versions of Laravel Lang packages into the application

2. Change the approach to documentation, bringing together all the packages and information about them in one place. I am going to work on this in the near future.

As for Laravel Lang Common, the project itself does not contain a single line of code - it only pulls the necessary versions of dependencies with it. Since several people wrote to me in private messages about installing on PHP 7.3 and 7.4, I decided to release three major versions at once:

1. `1.0` for PHP 7.3, 7.4, 8.0, 8.1 and Laravel/Lumen 7.x, 8.x, 9.x
2. `2.0` for PHP 8.0, 8.1 and Laravel/Lumen 7.x, 8.x, 9.x
3. `3.0` for PHP 8.1, 8.2 and Laravel/Lumen 8.x, 9.x, 10.x

Only the latest version will be kept up to date. The previous ones are for those who want to use our project on older versions of the framework and I did this only for their convenience.

Also, on the main page of the repository, I displayed a [compatibility table](https://github.com/Laravel-Lang/common#compatibility), looking at which you can immediately understand which version of the package is better to use for the application.

For example, we have Laravel 8.80 and PHP 7.4. The table says that we need to install the first version of `Common`:

```bash
composer require laravel-lang/common:^1.0
```

I think that with this step we increase loyalty to our products 😊

PS: @gamg, @MDylan, you will probably be interested too.

**Comment by ariaieboy** [comment]
A great addition to the organization, I was using laravel-lang v10 in one of my projects and I just forgot that laravel/Lang is in version 12.x. while in these 2 versions, we have no compatibility problem with newer projects that use the latest version of laravel and PHP. with this package we can manage all package versions for a given set of PHP and laravel versions in one place.

**Reply by andrey-helldar** [reply]
And, in addition, there is no need to manually open several projects and check the compatibility of versions.

**Reply by andrey-helldar** [reply]
PS: To check for updates, I use the `composer outdated` console command.

**Reply by ariaieboy** [reply]
> PS: To check for updates, I use the `composer outdated` console command.

I usually just update my packages using `composer update` and I use `composer outdated` or https://github.com/nunomaduro/patrol to check the outdated packages in my projects but I don't check the packages on regular bases I've found out about the outdated Laravel-Lang while I was working on a preparation PR for laravel 10 upgrade.

**Reply by andrey-helldar** [reply]
Yes, I use `composer update` all the time, and checking the availability of major versions is rare - about 1 time per month.

**Comment by mohamedsabil83** [comment]
It's a good idea. That will help many developers to fast get the proper version according to their projects. also, let old projects which used an older Laravel version take benefit from our packages.
